src/quickgui/plugin/editor/qgsquickexternalresource.qml

```diff
@@ -138,6 +138,13 @@ Item {
     }
   ]
 
+  onVisibleChanged: {
+    if (visible) {
+      image.currentValue = value
+      image.source = image.getSource()
+    }
+  }
+
```

Comment:
I am a bit worried this workaround may not work well - the change of the image source should not be dependent on the item visibility change. There is already image source being changed on 1. onCurrentValueChanged, 2. Component.onCompleted - so let's try to figure out the flow of events so we can have a more robust fix...